src/main/java/duke/task/Event.java

```diff
@@ -0,0 +1,51 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a type of Task that has a specific date and time of occurrence.
+ * Similar to Deadline Class.
+ */
+public class Event extends Task {
+    protected String at;
+    protected LocalDate date;
+    protected LocalTime time;
+
+    public Event(String description, String at) {
+        super(description);
+        this.at = at;
+        String[] timeComponent = at.split(" ");
+        this.date = LocalDate.parse(format(timeComponent[0].trim()));
```

Comment:
Since this is done for any task that is timed, perhaps this can be abstracted out into a parent class?